### ISSUE TITLE
Plugin crashes on terraform apply

### ISSUE THREAD

**Issue by niko-lay** [Telmate/terraform-provider-proxmox#466]
Hello
According to [docs](https://registry.terraform.io/providers/Telmate/proxmox/latest/docs), I've created new role `TerraformProv` in proxmox cluster, assign this role to the user `tfuser` and created test lxc container.
```
[12:33] [pm01 ~] # pveum role list  | grep TerraformProv
│ TerraformProv     │ Datastore.AllocateSpace,Datastore.Audit,VM.Allocate,VM.Audit,VM.Clone,VM.Config.CDROM,VM.Config.CPU,VM.Config.Cloudinit,VM.Config.Disk,VM.Config.HWType,VM.Config.Memory,VM.Config.Network,VM.Config.Options,VM.Monitor,VM.PowerMgmt    
[12:36] [pm01 ~] # pveum acl list | grep TerraformProv
│ /          │ TerraformProv │ user  │ tfuser@pve                       │ 1         │

[12:37] [pm01 ~] # cat test-lxc-01.tf
resource "proxmox_lxc" "basic" {
  target_node  = "pm01"
  hostname     = "lxc-basic-${var.env}"
  ostemplate   = "templates:vztmpl/ubuntu-20.04-standard_20.04-1_amd64.tar.gz"
  password     = random_string.container_root_passwd.result
  unprivileged = true
  protection   = true

  // Terraform will crash without rootfs defined
  rootfs {
    storage = "local-zfs"
    size    = "8G"
  }

  network {
    name   = "eth0"
    bridge = "vmbr1"
    ip     = "dhcp"
  }
}

resource "random_string" "container_root_passwd" {
    length = 10
}

[12:46] [pm01 ~] # pveversion 
pve-manager/6.4-13/9f411e79 (running kernel: 5.4.140-1-pve)

```
`terraform plan` works fine, but `terraform apply` created CT and crashed. 

```
random_string.container_root_passwd: Refreshing state... [id=xxxxxx]
proxmox_lxc.basic: Refreshing state... [id=pm01/lxc/241]
╷
│ Error: Plugin did not respond
│ 
│   with proxmox_lxc.basic,
│   on test-lxc-01.tf line 1, in resource "proxmox_lxc" "basic":
│    1: resource "proxmox_lxc" "basic" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-proxmox_v2.9.3 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 36 [running]:
github.com/Telmate/terraform-provider-proxmox/proxmox._resourceLxcRead(0xc00030c680, 0xbc5cc0, 0xc00046a000, 0x0, 0x0)
	github.com/Telmate/terraform-provider-proxmox/proxmox/resource_lxc.go:741 +0x1dd9
github.com/Telmate/terraform-provider-proxmox/proxmox.resourceLxcRead(0xc00030c680, 0xbc5cc0, 0xc00046a000, 0x128e6e0, 0xc000060c00)
	github.com/Telmate/terraform-provider-proxmox/proxmox/resource_lxc.go:658 +0x3f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc00043c8c0, 0xe0a8b8, 0xc000240c40, 0xc00030c680, 0xbc5cc0, 0xc00046a000, 0x0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:346 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00043c8c0, 0xe0a8b8, 0xc000240c40, 0xc000112680, 0xbc5cc0, 0xc00046a000, 0xc00000f7c8, 0x0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:635 +0x1cb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00009c0a8, 0xe0a8b8, 0xc000240c40, 0xc000240c80, 0xc000240c40, 0x40b965, 0xca2120)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/grpc_provider.go:576 +0x47d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc00004dbc0, 0xe0a960, 0xc000240c40, 0xc000548060, 0xc00004dbc0, 0xc000536030, 0xc0006e3ba0)
	github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/tf5server/server.go:298 +0x105
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0xcd0560, 0xc00004dbc0, 0xe0a960, 0xc000536030, 0xc000548000, 0x0, 0xe0a960, 0xc000536030, 0xc000144800, 0x3a1)
	github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000280540, 0xe12198, 0xc0006d2300, 0xc0000fc900, 0xc0003885a0, 0x1251dd0, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.42.0/server.go:1282 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000280540, 0xe12198, 0xc0006d2300, 0xc0000fc900, 0x0)
	google.golang.org/grpc@v1.42.0/server.go:1616 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0000a0350, 0xc000280540, 0xe12198, 0xc0006d2300, 0xc0000fc900)
	google.golang.org/grpc@v1.42.0/server.go:921 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.42.0/server.go:919 +0x1fd

Error: The terraform-provider-proxmox_v2.9.3 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```
If I add acl for `tfuser` to `PVEAdmin` then  `terraform apply` works fine. 
Looks like it is not enough permissions for role `TerraformProv`. Could you please clarify, which permissions should be added to role `TerraformProv`?

**Comment by rkarpuzov** [comment]
In my opinion there is no permissions issue. I have the same problem with qemu VMs:

proxmox_vm_qemu.proxmox_vm: Still creating... [10s elapsed]
╷
│ Error: Plugin did not respond
│ 
│   with proxmox_vm_qemu.proxmox_vm,
│   on main.tf line 26, in resource "proxmox_vm_qemu" "proxmox_vm":
│   26: resource "proxmox_vm_qemu" "proxmox_vm" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-proxmox_v2.9.3 plugin:

panic: interface conversion: interface {} is nil, not []interface {}

goroutine 64 [running]:
github.com/Telmate/proxmox-api-go/proxmox.(*Client).GetVmRefsByName(0xc0005a2eb0, 0xc0003aa510, 0xb, 0x1, 0x1, 0x0, 0x0, 0xc0001e8a00)
        github.com/Telmate/proxmox-api-go@v0.0.0-20211123192920-062fd1a6ab10/proxmox/client.go:190 +0x845
github.com/Telmate/proxmox-api-go/proxmox.(*Client).GetVmRefByName(...)
        github.com/Telmate/proxmox-api-go@v0.0.0-20211123192920-062fd1a6ab10/proxmox/client.go:180
github.com/Telmate/terraform-provider-proxmox/proxmox.resourceVmQemuCreate(0xc0003c5300, 0xbc5cc0, 0xc0005a2f00, 0xa, 0xffffffffffffffff)
        github.com/Telmate/terraform-provider-proxmox/proxmox/resource_vm_qemu.go:738 +0x1886
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc00036c460, 0xe0a8b8, 0xc0000d3b40, 0xc0003c5300, 0xbc5cc0, 0xc0005a2f00, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:329 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00036c460, 0xe0a8b8, 0xc0000d3b40, 0xc0002b7110, 0xc0003c5180, 0xbc5cc0, 0xc0005a2f00, 0x0, 0x0, 0x0, ...)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:467 +0x67b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0002dc0c0, 0xe0a8b8, 0xc0000d3b40, 0xc0002af0e0, 0xc0000d3b40, 0xcd0560, 0xc0003b5e00)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/grpc_provider.go:977 +0xacf
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000115d00, 0xe0a960, 0xc0000d3b40, 0xc0005c4fc0, 0xc000115d00, 0xc0003b5e90, 0xc0005b5ba0)
        github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/tf5server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0xcd0560, 0xc000115d00, 0xe0a960, 0xc0003b5e90, 0xc0005ab020, 0x0, 0xe0a960, 0xc0003b5e90, 0xc000495000, 0xe24)
        github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002f4540, 0xe12198, 0xc000001680, 0xc0004907e0, 0xc00022bdd0, 0x1251e00, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.42.0/server.go:1282 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0002f4540, 0xe12198, 0xc000001680, 0xc0004907e0, 0x0)
        google.golang.org/grpc@v1.42.0/server.go:1616 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0000b83f0, 0xc0002f4540, 0xe12198, 0xc000001680, 0xc0004907e0)
        google.golang.org/grpc@v1.42.0/server.go:921 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.42.0/server.go:919 +0x1fd

Error: The terraform-provider-proxmox_v2.9.3 plugin crashed!

Curiously, after unexpectedly switched off network, seems to work:
 proxmox_vm_qemu.proxmox_vm: Creating...
proxmox_vm_qemu.proxmox_vm: Still creating... [10s elapsed]
proxmox_vm_qemu.proxmox_vm: Still creating... [20s elapsed]
proxmox_vm_qemu.proxmox_vm: Still creating... [30s elapsed]
^C
Interrupt received.

**Comment by polarlights** [comment]
terraform-plugin-proxmox.log may help u.

**Comment by niko-lay** [comment]
> terraform-plugin-proxmox.log may help u.

@polarlights thank you for suggestion, it helped. 

`Sys.audit` and `Pool.allocate` permissions should be added to the role `TerraformProv`

**Comment by Houssem12-ai** [comment]
i have the same issue also 
i'am trying to create vm proxmox servers remotely from my local computer
 
proxmox_vm_qemu.test_server[0]: Creating...
╷
│ Error: Plugin did not respond
│ 
│   with proxmox_vm_qemu.test_server[0],
│   on main.tf line 21, in resource "proxmox_vm_qemu" "test_server":
│   21: resource "proxmox_vm_qemu" "test_server" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-proxmox_v2.9.3 plugin:

panic: interface conversion: interface {} is nil, not []interface {}

goroutine 56 [running]:
github.com/Telmate/proxmox-api-go/proxmox.(*Client).GetVmRefsByName(0xc0000d0ff0, 0xc000034300, 0x9, 0x1, 0x1, 0x0, 0x0, 0xc0001c2000)
        github.com/Telmate/proxmox-api-go@v0.0.0-20211123192920-062fd1a6ab10/proxmox/client.go:190 +0x845
github.com/Telmate/proxmox-api-go/proxmox.(*Client).GetVmRefByName(...)
        github.com/Telmate/proxmox-api-go@v0.0.0-20211123192920-062fd1a6ab10/proxmox/client.go:180
github.com/Telmate/terraform-provider-proxmox/proxmox.resourceVmQemuCreate(0xc000184f80, 0xbc5cc0, 0xc0000d1040, 0x0, 0xffffffffffffffff)
        github.com/Telmate/terraform-provider-proxmox/proxmox/resource_vm_qemu.go:738 +0x1886
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0003c6460, 0xe0a8b8, 0xc000208c40, 0xc000184f80, 0xbc5cc0, 0xc0000d1040, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:329 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0003c6460, 0xe0a8b8, 0xc000208c40, 0xc000506a90, 0xc000184d00, 0xbc5cc0, 0xc0000d1040, 0x0, 0x0, 0x0, ...)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:467 +0x67b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0000b40a8, 0xe0a8b8, 0xc000208c40, 0xc000234370, 0xc000208c40, 0xcd0560, 0xc000566200)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/grpc_provider.go:977 +0xacf
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0005433a0, 0xe0a960, 0xc000208c40, 0xc00016a070, 0xc0005433a0, 0xc0005662d0, 0xc000272ba0)
        github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/tf5server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0xcd0560, 0xc0005433a0, 0xe0a960, 0xc0005662d0, 0xc000110a80, 0x0, 0xe0a960, 0xc0005662d0, 0xc000344d80, 0xce6)
        github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000298540, 0xe12198, 0xc000001380, 0xc000128360, 0xc000207d40, 0x1251e00, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.42.0/server.go:1282 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000298540, 0xe12198, 0xc000001380, 0xc000128360, 0x0)
        google.golang.org/grpc@v1.42.0/server.go:1616 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00010a440, 0xc000298540, 0xe12198, 0xc000001380, 0xc000128360)
        google.golang.org/grpc@v1.42.0/server.go:921 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.42.0/server.go:919 +0x1fd

Error: The terraform-provider-proxmox_v2.9.3 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

**Comment by rkarpuzov** [comment]
> i have the same issue also i'am trying to create vm proxmox servers remotely from my local computer
> 
> proxmox_vm_qemu.test_server[0]: Creating... ╷ │ Error: Plugin did not respond │ │ with proxmox_vm_qemu.test_server[0], │ on main.tf line 21, in resource "proxmox_vm_qemu" "test_server": │ 21: resource "proxmox_vm_qemu" "test_server" { │ │ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details. ╵
> 
> Stack trace from the terraform-provider-proxmox_v2.9.3 plugin:
> 
> panic: interface conversion: interface {} is nil, not []interface {}

 <cut>
> 
> Error: The terraform-provider-proxmox_v2.9.3 plugin crashed!
> 
> This is always indicative of a bug within the plugin. It would be immensely helpful if you could report the crash with the plugin's maintainers so that it can be fixed. The output above should help diagnose the issue.

Double check your rights. I see these errors only if I have wrong permissions. Such as type of login (pve or pam), not fully translated ! sign, if you use bash var PM_USER or PM_API_TOKEN_ID, you mus export it (and may store it in your .bashrc file) like `exportPM_API_TOKEN_ID="user\@pve\!terraform"` or use single quotes instead of double quotes.

**Comment by brandon-welsch** [comment]
I had the same problems and everything was solved. Thanks @Houssem12-ai @rkarpuzov

**Comment by Houssem12-ai** [comment]
@brandon-welsch what have you did to solve the problem ?

**Comment by alteredtech** [comment]
What does your providers.tf. I was able to get this working. With the debugging added to the provider, I was able to realize that it wasnt making connection to the server. what I was missing is the 's' in https even though tls_insecure is set to true. Once I was able to do that and change the node from ip to the host name (I wish that was better explained, I might try doing my first merge request?)
